tap_amazon_ads/exceptions.py

```diff
@@ -0,0 +1,136 @@
+class Amazon_AdsError(Exception):
+    """class representing Generic Http error."""
+
+    def __init__(self, message=None, response=None):
+        try:
+            super().__init__(message)
+            self.message = message
+            self.response = response
+        except (IndexError, AttributeError):
+            pass
```

Comment:
The constructor for `Amazon_AdsError` catches and passes on exceptions, but it’s unclear when this would be triggered. Consider removing unless you know of a specific edge case.
```suggestion
    def __init__(self, message=None, response=None):
        super().__init__(message)
        self.message = message
        self.response = response
```

Reply:
Addressed review comment.